compiler-plugin/src/main/java/io/ballerina/stdlib/nats/plugin/NatsServiceValidator.java

```diff
@@ -95,42 +94,48 @@ private void validateAttachPoint(SyntaxNodeAnalysisContext context) {
+                    // Case 2: Service name is not a string and no annotation
+                    reportError(context, CompilationErrors.INVALID_SERVICE_ATTACH_POINT, serviceDeclarationNode);
                 }
-            } else {
-                if (attachPoint.get().kind() != ServiceAttachPointKind.STRING_LITERAL) {
-                    if (serviceDeclarationSymbol.annotations().isEmpty()) {
-                        context.reportDiagnostic(PluginUtils.getDiagnostic(
-                                CompilationErrors.INVALID_SERVICE_ATTACH_POINT,
-                                DiagnosticSeverity.ERROR, serviceDeclarationNode.location()));
-                    } else {
-                        validateAnnotation(symbolList.get(0), serviceDeclarationNode.location(),
-                                context);
-                    }
+            } else if (!hasServiceConfig(annotations)) {
+                if (serviceNameAttachPoint.isEmpty()) {
+                    // Case 1: No service name and no annotation
+                    reportError(context, CompilationErrors.NO_ANNOTATION, serviceDeclarationNode);
+                } else if (serviceNameAttachPoint.get().kind() != ServiceAttachPointKind.STRING_LITERAL) {
+                    // Case 2: Service name is not a string and no annotation
+                    reportError(context, CompilationErrors.INVALID_SERVICE_ATTACH_POINT, serviceDeclarationNode);
                 }
             }
         }
     }
 
-    private void validateAnnotation(AnnotationSymbol annotationSymbol, Location location,
-                                    SyntaxNodeAnalysisContext context) {
-        Optional<ModuleSymbol> moduleSymbolOptional = annotationSymbol.getModule();
-        ModuleSymbol moduleSymbol = moduleSymbolOptional.get();
-        if (!moduleSymbol.id().orgName().equals(PluginConstants.PACKAGE_ORG) ||
-                !moduleSymbol.id().moduleName().equals(PluginConstants.PACKAGE_PREFIX)) {
-            context.reportDiagnostic(PluginUtils.getDiagnostic(CompilationErrors.INVALID_ANNOTATION,
-                    DiagnosticSeverity.ERROR, location));
+    private boolean hasServiceConfig(List<AnnotationSymbol> annotationSymbols) {
+        boolean flag = false;
+        for (AnnotationSymbol annotationSymbol : annotationSymbols) {
+            Optional<ModuleSymbol> moduleSymbolOptional = annotationSymbol.getModule();
+            if (moduleSymbolOptional.isPresent()) {
+                ModuleSymbol moduleSymbol = moduleSymbolOptional.get();
+                if (moduleSymbol.id().orgName().equals(PluginConstants.PACKAGE_ORG) ||
+                        moduleSymbol.id().moduleName().equals(PluginConstants.PACKAGE_PREFIX)) {
+                    // not checking name as nats has only two annotations and only one is allowed on services.
+                    flag = true;
+                }
+            }
         }
+        return flag;
```

Comment:
Check [this comment](https://github.com/ballerina-platform/module-ballerinax-rabbitmq/pull/874#discussion_r1298299280)

Reply:
fixed in 3808e110f7b804b6fd363d08c8b433c1e31edfef